src/sync/sync_permission.cpp

```diff
@@ -61,6 +61,37 @@ Permission::AccessLevel extract_access_level(Object& permission, CppContext& con
 
 }
 
+// MARK: - Permission
+
+std::string Permission::description_for_access_level(AccessLevel level)
+{
+    switch (level) {
+        case AccessLevel::None: return "none";
+        case AccessLevel::Read: return "read";
+        case AccessLevel::Write: return "write";
+        case AccessLevel::Admin: return "admin";
+    }
+    REALM_UNREACHABLE();
+}
+
+bool Permission::paths_are_equivalent(std::string path_1, std::string path_2,
+                                      std::string user_id_1, std::string user_id_2)
+{
+    if (path_1 == path_2) {
+        return true;
```

Comment:
Oh, that's a good point. In the original Objective-C code we checked user IDs before calling this method, but we shouldn't assume the same will be true for this public C++ API.